pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -179,8 +181,30 @@ func (c *client) UpdateFunction(ctx context.Context, fm FunctionManifest) error
 		return fmt.Errorf("unknown error given: %w", err)
 	}
 
-	// TODO: Support more configurable fields using Lambda.UpdateFunctionConfiguration.
-	// https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#UpdateFunctionConfiguration
+	retry := backoff.NewRetry(RequestRetryTime, backoff.NewConstant(RetryIntervalDuration))
+	updateFunctionConfigurationSucceed := false
+	var configErr error
+	for retry.WaitNext(ctx) {
+		configInput := &lambda.UpdateFunctionConfigurationInput{
+			FunctionName: aws.String(fm.Spec.Name),
+			MemorySize:   aws.Int64(fm.Spec.Memory),
+			Timeout:      aws.Int64(fm.Spec.Timeout),
+			Environment: &lambda.Environment{
+				Variables: aws.StringMap(fm.Spec.Environments),
+			},
```

Comment:
These fields too.

Reply:
in case `fm.Spec.Environments` is nil, all environment variables will be removed from the remote lambda function configuration. So I think we don't need to validate the nullity of `fm.Spec.Environments`
About, `Timeout` is discussed on https://github.com/pipe-cd/pipe/pull/1548#discussion_r570022638 thread 🙏

Reply:
For users who migrate their already running Lambdas to use PipeCD,
at the first time, this `Spec.Environments` is nil and all pre-configured environments will be removed.
Is that right?
So I think we should set this field only if it is not nil to avoid that case.
(And the difference can be detected by drift-detector in the future.)

Reply:
In that cae, I think users will can't remove environment vairiables via function.yaml 😅 
To keep simplicity and make remote configuration as same as function.yaml config, keep it as is is good to me. How do you think about this 👀

Reply:
Ok. Got your point. So let's add a note about that to the document later.

Reply:
Sure, thank you 😁